### PR TITLE
Support Testcontainers framework in junit testings

### PR DESCRIPTION
**Pull request by lucyge2022**
### What changes are proposed in this pull request?

Add testcontainers test support for relevant unit testings.

### Why are the changes needed?

For certain junit test suites such as etcd related, testcontainers framework needs to be supported

### Does this PR introduce any user facing changes?

N/A

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -858,7 +858,13 @@
       <dependency>
         <groupId>org.testcontainers</groupId>
         <artifactId>testcontainers</artifactId>
-        <version>1.14.3</version>
+        <version>1.18.3</version>
```
Comment (apc999) [inline pom.xml:861]: nit: let's define `${testcontainers.version}` to be `1.18.3` and use the var here, so we can ensure both artifact are in the same version

File: dora/testcontainers-tests/src/test/java/alluxio/membership/MembershipManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+  private static final int ETCD_PORT = 2379;
+  @Rule
+  public TemporaryFolder mFolder = new TemporaryFolder();
+
+  private static ToxiproxyContainer.ContainerProxy etcdProxy;
+
+  //Add for logging for debugging purpose
+  @BeforeClass
+  public static void init() {
+    PropertyConfigurator.configure("/Users/lucyge/Documents/github/alluxio/conf/log4j.properties");
```
Comment (apc999) [inline dora/testcontainers-tests/src/test/java/alluxio/membership/MembershipManagerTest.java:65]: shall we use a rand path from tempfolder? any issue with a hardcoded path here?

File: dora/testcontainers-tests/pom.xml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.alluxio</groupId>
+        <artifactId>alluxio-dora</artifactId>
+        <version>303-SNAPSHOT</version>
+    </parent>
+    <artifactId>alluxio-testcontainers-tests</artifactId>
```
Comment (apc999) [inline dora/testcontainers-tests/pom.xml:22]: ideally, we shall have this in `dora/tests/container` rather than a top-level module

**Review by apc999 — CHANGES_REQUESTED** [review]

File: dora/testcontainers-tests/src/test/java/alluxio/membership/MembershipManagerTest.java (hunk 2 above)
Comment (lucyge2022) [reply dora/testcontainers-tests/src/test/java/alluxio/membership/MembershipManagerTest.java]: sorry this is a local change checked in, will remove it as part of final readiness

File: pom.xml (hunk 1 above)
Comment (lucyge2022) [reply pom.xml:861]: done

File: dora/testcontainers-tests/pom.xml (hunk 3 above)
Comment (lucyge2022) [reply dora/testcontainers-tests/pom.xml]: done

File: .github/workflows/java8_container_tests.yml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  build:
+    name: "modules: "
+
+    strategy:
+      fail-fast: false
+      matrix:
+        modules:
+          - >-
+            alluxio.membership.**
+    # TODO: update the above
```
Comment (Xenorith) [inline .github/workflows/java8_container_tests.yml:18]: remove TODO comment

File: dora/tests/testcontainers/src/test/java/alluxio/membership/MembershipManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+  //Uncomment for logging when need debugging
+  /*
+  @BeforeClass
+  public static void init() {
+    PropertyConfigurator.configure("alluxio/conf/log4j.properties");
+    Properties props = new Properties();
+    props.setProperty(PropertyKey.LOGGER_TYPE.toString(), "Console");
+  }
+  */
```
Comment (Xenorith) [inline dora/tests/testcontainers/src/test/java/alluxio/membership/MembershipManagerTest.java:64]: i don't think it's customary to leave these types of commented blocks? but ignore me if it is

File: .github/workflows/java8_container_tests.yml (hunk 4 above)
Comment (lucyge2022) [reply .github/workflows/java8_container_tests.yml:18]: done

File: dora/tests/testcontainers/pom.xml.bak
Comment (Xenorith) [inline dora/tests/testcontainers/pom.xml.bak:1]: remove this file

File: pom.xml [hunk 6]
```diff
@@ -858,7 +859,13 @@
       <dependency>
         <groupId>org.testcontainers</groupId>
         <artifactId>testcontainers</artifactId>
-        <version>1.14.3</version>
+        <version>${testcontainers.version}</version>
```
Comment (Xenorith) [inline pom.xml:862]: just double checking that previously it was 1.14.3 and now it is upgraded to 1.18.3

**Review by apc999 — APPROVED** [review]

File: dora/tests/testcontainers/pom.xml.bak
Comment (lucyge2022) [reply dora/tests/testcontainers/pom.xml.bak:1]: done

**Review by Xenorith — APPROVED** [review]

File: pom.xml (hunk 6 above)
Comment (lucyge2022) [reply pom.xml:862]: yeah so I can confirm the only places using testcontainer lib is from this line:

```
dora/tests/src/test/java/alluxio/server/worker/WorkerMetadataSyncIntegrationTest.java:28:
import org.testcontainers.shaded.com.google.common.collect.ImmutableMap;
```
which is totally unnecessary and could be independent of which version of testcontainer  we have originally

**Comment by lucyge2022** [comment]
alluxio-bot, merge this please.

**Pull request merged**